library/src/main/java/com/firebase/ui/FirebaseLoginBaseActivity.java

```diff
@@ -90,6 +128,38 @@ public void onError(Exception ex) {
                 // TODO: Raise GMS Dialog Box?
             }
         });
+
+        mTwitterAuthHelper = new TwitterAuthHelper(this, new TokenAuthHandler() {
+            @Override
+            public void onTokenReceived(FirebaseOAuthToken token) {
+                authenticateRefWithFirebaseOAuthToken(token);
+            }
+
+            @Override
+            public void onCancelled() {
+                onFirebaseLoginCancel();
+            }
+
+            @Override
+            public void onError(Exception ex) {
+                // TODO: Raise GMS Dialog Box?
+            }
+        });
+    }
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Log.d("BASE", "ACTIVITY" + data.toString());
+        mFacebookAuthHelper.mCallbackManager.onActivityResult(requestCode, resultCode, data);
+        mTwitterAuthHelper.onActivityResult(requestCode, resultCode, data);
+    }
+
+    public void showFirebaseLoginPrompt() {
+        mDialog = new FirebaseLoginDialog();
+        mDialog
```

Comment:
We should allow the user that derives from the BaseActivity to specify what providers they want enabled.